spec/controllers/chapters_controller_spec.rb

```diff
@@ -460,6 +460,15 @@
             expect(assigns[:chapter].posted).to be true
           end
 
+          it "updates cached chapter counts" do
+            expect do
+              post :create, params: { work_id: work.id, chapter: chapter_attributes, post_without_preview_button: true }
+            end.to change { work.number_of_chapters }
+              .from(1).to(2)
+              .and change { work.number_of_posted_chapters }
```

Comment:
Should we `reload` the work? It seems like we always use that with `avoid_changing`.

Reply:
In this specific case, `work` doesn't need reloading to reflect the changes. `number_of_chapters` and `number_of_posted_chapters` are set up with cache keys that only include the work's ID (so reloading the work won't change the cache key), and the contents of the cache are set using `.count` (which forces a DB query even if the association is already loaded).